src/webgpu/shader/execution/builtin/builtin.ts

```diff
@@ -73,94 +297,89 @@ export function runShaderTestImpl<F extends NumberType>(
-        inputValue.push(formatNum(inputData[idx]));
-
-        // `cases[expectedIndex].expected` is an array of values that are treated as a pass.
-        // Do any of these expected values match?
-        const elementMatched = cases[expectedIndex].expected.some(e => e.value === outputData[idx]);
-        outputValue.push(formatNum(outputData[idx], elementMatched ? Colors.green : Colors.red));
-
-        const caseExpected: string[] = [];
-        for (const e of cases[expectedIndex].expected) {
-          if (outputData[idx] === e.value) {
-            // This expected value matched
-            caseExpected.push(formatNum(e.value, Colors.green));
-          } else if (elementMatched) {
-            // The element matched, but it wasn't for this value.
-            caseExpected.push(formatNum(e.value, Colors.grey));
-          } else {
-            // The element did not match any expected value
-            caseExpected.push(formatNum(e.value, Colors.red));
-          }
-        }
-        expectedValue.push(caseExpected.join(' or '));
-
-        // If none of the expected values matched, then the case has failed.
-        caseMatched = caseMatched && elementMatched;
-      }
-
-      if (!caseMatched) {
-        if (arrayLength > 1) {
-          errs.push(
-            `${builtin}(${type}(${inputValue.join(', ')}))\n` +
-              `    returned: ${type}(${outputValue.join(', ')})\n` +
-              `    expected: ${type}(${expectedValue.join(', ')})`
-          );
-        } else {
-          errs.push(
-            `${builtin}(${inputValue.join(', ')})\n` +
-              `    returned: ${outputValue.join(', ')}\n` +
-              `    expected: ${expectedValue.join(', ')}`
-          );
-        }
+    for (let caseIdx = 0; caseIdx < cases.length; caseIdx++) {
+      const c = cases[caseIdx];
+      const got = outputs[caseIdx];
+      const cmp = toComparator(c.expected)(got);
+      if (!cmp.matched) {
+        errs.push(`${builtin}(${c.input instanceof Array ? c.input.join(', ') : c.input})
+    returned: ${cmp.got}
+    expected: ${cmp.expected}`);
       }
     }
 
-    if (errs.length > 0) {
-      return new Error(errs.join('\n\n'));
-    }
-    return undefined;
+    return errs.length > 0 ? new Error(errs.join('\n\n')) : undefined;
   };
 
   t.expectGPUBufferValuesPassCheck(outputBuffer, checkExpectation, {
-    type: arrayType,
-    typedLength: inputData.length,
+    type: Uint8Array,
+    typedLength: outputBufferSize,
   });
 }
 
+// Packs a list of scalar test cases into a smaller list of vector cases.
+export function packScalarsToVector(cases: CaseList, vectorWidth: number): CaseList {
+  const asScalar = function (val: Array<Value> | Value): Scalar {
+    if (val instanceof Scalar) {
+      return val;
+    }
+    if (val instanceof Array && val.length === 1) {
+      return asScalar(val[0]);
+    }
+    throw new Error('packScalarsToVector() requires all Case values to be a single scalar Value');
```

Comment:
So if I understand correctly, this is not intended to work with a builtin function that takes multiple arguments, because the input field would be an array of scalars.
Example:     `select` would have input example of [number, number, bool] which is an array of scalars.

That's ok, but not super clear.  Maybe an updated comment at 331?

Reply:
Done